### ISSUE TITLE
New develop branch manual install with yarn giving error

### ISSUE THREAD

**Issue by Alex-YddeR** [edumeet/edumeet#770]
Dear Team,

New develop branch manual install with yarn giving error
00h00m00s 0/0: : ERROR: There are no scenarios; must have at least one.

getting below error when starting server with **node server.js** or pm2 tool

import Logger from './lib/Logger';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
ubuntu@ip-10-0-3-85:~/edumeet/server$ 

Thanks

**Comment by vpalmisano** [comment]
Pull the latest develop revision and use `yarn start` or `yarn dev` command.

**Comment by vpalmisano** [comment]
You are using the wrong package, see here: https://stackoverflow.com/questions/53471063/yarn-error-there-are-no-scenarios-must-have-at-least-one/53471064

**Comment by Alex-YddeR** [comment]
Fixed yarn with **npm install --global yarn** , but getting error visiting webpage as below

react-dom.production.min.js:290 Uncaught Error: Minified React error #200; visit https://reactjs.org/docs/error-decoder.html?invariant=200 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at t.render (react-dom.production.min.js:290)
    at index.js:180
    at index.js:75
    at HTMLDocument.e (ready.js:23)
t.render @ react-dom.production.min.js:290
(anonymous) @ index.js:180
(anonymous) @ index.js:75
e @ ready.js:23
:4443/favicon.ico:1 GET https://test.alex.com.com:4443/favicon.ico 500

**Comment by vpalmisano** [comment]
The yarn install procedure is:
```
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list

sudo apt update
sudo apt install yarn
```

**Comment by Alex-YddeR** [comment]
Same issue i tried in fresh machine ubuntu 18.04. Above steps of yarn installation works only when Nodejs 14.X is not installed. 

Same error on browser as below

 react-dom.production.min.js:290 Uncaught Error: Minified React error #200; visit https://reactjs.org/docs/error-decoder.html?invariant=200 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at t.render (react-dom.production.min.js:290)
    at index.js:180
    at index.js:75
    at HTMLDocument.e (ready.js:23)
t.render @ react-dom.production.min.js:290
(anonymous) @ index.js:180
(anonymous) @ index.js:75
e @ ready.js:23
favicon.ico:1 GET https://test.alex.com:4443/favicon.ico 500

Thanks

**Comment by vpalmisano** [comment]
Do you have tried to remove the node_modules folder first?

**Comment by Alex-YddeR** [comment]
Yes, its completely fresh manual installation and tried again its the same issue its getting failed in one or other place.

If everything is installed web portal is not working
First of all yarn not going to install when node version 14 is already present, so we need to install first yarn and then install nodejs.

**Comment by Astagor** [comment]
This is a problem, as how to upgrade existing installations?

**Comment by vpalmisano** [comment]
@Alex-YddeR can you provide us the exact nodejs and yarn packages versions?

**Comment by Alex-YddeR** [comment]
Platform: 
Ubuntu 18.04 / 20.04
Install Steps:

1. Install Yarn
> sudo apt-get update
> curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
> echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
> sudo apt update
> sudo apt install yarn ( Install default node 8.10.0)
or
> sudo apt install --no-install-recommends yarn

**Yarn Version : v1.22.5** 


2. Install dependencies
> sudo apt install git npm yarnpkg build-essential redis libssl-dev openssl pkg-config

**Gives error on installing yarnpkg here.** so skipped yarnpkg


3. Install NodeJS
# Using Ubuntu
> curl -fsSL https://deb.nodesource.com/setup_lts.x | sudo -E bash -
> sudo apt-get install -y nodejs
**Node Version: 14.16.1**

yarn works well building the app, it either fails on server or everything installs well but web portal doesn't work 

Thanks

**Comment by Alex-YddeR** [comment]
Some how we tried to install, but when try to reach the page by adding room name it says internal server error. also when exit the conference with leave button same error is coming as per below screenshot

![Screenshot 2021-04-15 at 4 57 34 PM](https://user-images.githubusercontent.com/65006646/114862305-fb20ed00-9e0b-11eb-8486-946b1cde619c.png)

Thanks

**Comment by vpalmisano** [comment]
Can you provide also the build log of server and app?

**Comment by Alex-YddeR** [comment]
Sure, where can I get those logs
Thanks

**Comment by Alex-YddeR** [comment]
Are you looking at below logs



ubuntu@ip-10-0-3-38:~/edumeet/app$ ls
Procfile  build  package.json  public  src  tsconfig.json  yarn.lock
ubuntu@ip-10-0-3-38:~/edumeet/app$ yarn
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @typescript-eslint/eslint-plugin@4.20.0" has unmet peer dependency "eslint@^5.0.0 || ^6.0.0 || ^7.0.0".
warning " > @typescript-eslint/parser@4.20.0" has unmet peer dependency "eslint@^5.0.0 || ^6.0.0 || ^7.0.0".
warning " > babel-eslint@10.1.0" has unmet peer dependency "eslint@>= 4.12.1".
warning " > eslint-config-react-app@6.0.0" has unmet peer dependency "eslint@^7.5.0".
warning " > eslint-plugin-flowtype@5.4.0" has unmet peer dependency "eslint@^7.1.0".
warning " > eslint-plugin-import@2.22.1" has unmet peer dependency "eslint@^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0".
warning " > eslint-plugin-jest@24.3.3" has unmet peer dependency "eslint@>=5".
warning " > eslint-plugin-jsx-a11y@6.4.1" has unmet peer dependency "eslint@^3 || ^4 || ^5 || ^6 || ^7".
warning " > eslint-plugin-react@7.23.1" has unmet peer dependency "eslint@^3 || ^4 || ^5 || ^6 || ^7".
warning " > eslint-plugin-react-hooks@4.2.0" has unmet peer dependency "eslint@^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0".
warning " > eslint-webpack-plugin@2.5.3" has unmet peer dependency "eslint@^7.0.0".
warning " > eslint-webpack-plugin@2.5.3" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning " > riek@1.1.0" has incorrect peer dependency "react@^0.14.0 || ^15.5.4".
warning " > riek@1.1.0" has incorrect peer dependency "react-dom@^0.14.0 || ^15.5.4".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb-base@14.2.1" has unmet peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
warning " > eslint-config-prettier@8.1.0" has unmet peer dependency "eslint@>=7.0.0".
warning " > eslint-import-resolver-typescript@2.4.0" has unmet peer dependency "eslint@*".
warning " > eslint-plugin-prettier@3.3.1" has unmet peer dependency "eslint@>=5.0.0".
[4/4] Building fresh packages...
Done in 51.88s.
ubuntu@ip-10-0-3-38:~/edumeet/app$ yarn build
yarn run v1.22.5
$ react-scripts build && mkdir -p ../server/public && rm -rf ../server/public/* && cp -r build/* ../server/public/
Creating an optimized production build...
`	Compiled with warnings.

src/components/JoinDialog.js
  Line 208:8:  '_askForPerms' is assigned a value but never used  no-unused-vars

src/components/Loader/LoadingView.js
  Line 4:17:  'theme' is defined but never used  no-unused-vars

src/components/MeetingViews/Filmstrip.js
  Line 265:4:  'roomClient' is assigned a value but never used  no-unused-vars

src/components/PeerAudio/PeerAudio.js
  Line 103:5:  'dst' is assigned a value but never used  no-unused-vars

src/components/VideoContainers/FullScreenView.js
  Line 173:5:  React Hook useEffect has missing dependencies: 'consumer' and 'roomClient'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

src/store.js
  Line 66:7:  'saveSubsetFilter' is assigned a value but never used  no-unused-vars

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

File sizes after gzip:

  212.98 KB  build/static/js/11.ff61e6df.chunk.js
  90.94 KB   build/static/js/main.4f42e8e2.chunk.js
  64.65 KB   build/static/js/13.9f5cd9fc.chunk.js
  64.03 KB   build/static/js/12.f17f29ca.chunk.js
  29.41 KB   build/static/js/room.36e6b750.chunk.js
  21.11 KB   build/static/js/1.a568a4a8.chunk.js
  18.49 KB   build/static/js/14.c4c15352.chunk.js
  6.73 KB    build/static/js/0.501fcd1f.chunk.js
  1.77 KB    build/static/js/runtime-main.53f61051.js
  1.28 KB    build/static/js/file-saver.99a4fcde.chunk.js
  1.09 KB    build/static/js/screensharing.3558591f.chunk.js
  720 B      build/static/css/12.f94bdb3a.chunk.css
  421 B      build/static/js/app.1300cd1f.chunk.js
  357 B      build/static/css/main.ea979dc9.chunk.css
  203 B      build/static/js/webtorrent.3ed9e320.chunk.js
  155 B      build/static/js/createtorrent.2bb026e9.chunk.js
  141 B      build/static/js/socket.io.d6da3eb8.chunk.js

The project was built assuming it is hosted at ./.
You can control this with the homepage field in your package.json.

The build folder is ready to be deployed.

Find out more about deployment here:

  https://cra.link/deployment

Done in 80.65s.

ubuntu@ip-10-0-3-38:~/edumeet/app$ cd ../server/
ubuntu@ip-10-0-3-38:~/edumeet/server$ ls
access.ts  certs  config  connect.js  dist  httpHelper.js  lib  node_modules  package.json  permissions.js  public  server.js  tsconfig.json  userRoles.js  utils  yarn.lock
ubuntu@ip-10-0-3-38:~/edumeet/server$ rm -rf node_modules/
ubuntu@ip-10-0-3-38:~/edumeet/server$ yarn
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-jsx-a11y@^6.4.1".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react@^7.21.5".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has unmet peer dependency "eslint-plugin-react-hooks@^4 || ^3 || ^2.3.0 || ^1.7.0".
warning " > eslint-config-prettier@8.1.0" has incorrect peer dependency "eslint@>=7.0.0".
[4/4] Building fresh packages...
Done in 252.57s.


ubuntu@ip-10-0-3-38:~/edumeet/server$ yarn build
yarn run v1.22.5
$ rm -rf dist && tsc && ln -s ../certs dist/certs
Done in 17.62s.

> ubuntu@ip-10-0-3-38:~/edumeet/server$ sudo node dist/server.js
- process.env.DEBUG: undefined
- config.mediasoup.worker.logLevel: debug
- config.mediasoup.worker.logTags: [ 'info', 'ice', 'dtls', 'rtp', 'srtp', 'rtcp' ]
(node:8686) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
(Use `node --trace-deprecation ...` to show where the warning was created)

**Comment by Alex-YddeR** [comment]
Tried to install again on fresh machine we are getting same error when tried to refresh the page or by adding roomname to URL. below is the error

Internal Server Error
If you report this error, please also report this tracking ID which makes it possible to locate your session in the logs which are available to the system administrator: ce005d47-e6ab-40b1-8109-bd288c35769a

Thanks

**Comment by Alex-YddeR** [comment]
@vpalmisano This is fixed by commit #782 

Thanks
